### PR TITLE
crimson: pickup change to fix --cpuset support and cleanups

### PR DESCRIPTION
**Pull request by tchaikov**
## Checklist
- [ ] References tracker ticket
- [ ] Updates documentation if necessary
- [ ] Includes tests for new functionality or reproducer for bug

---

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard backend`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`

</details>

### CODE REVIEW

**Comment by tchaikov** [comment]
@liu-chunmei

**Comment by tchaikov** [comment]
jenkins test crimson perf

**Comment by tchaikov** [comment]
jenkins test make check

File: src/crimson/thread/Condition.h [hunk 1]
```diff
@@ -1,36 +0,0 @@
-// -*- mode:C++; tab-width:8; c-basic-offset:2; indent-tabs-mode:t -*-
```
Comment (rzarzynski) [inline src/crimson/thread/Condition.h]: It's good to remove SLOCs but it's even better to drop entire files. :-)

File: src/crimson/CMakeLists.txt [hunk 2]
```diff
@@ -8,6 +8,7 @@ if((CMAKE_CXX_COMPILER_ID STREQUAL GNU AND CMAKE_CXX_COMPILER_VERSION VERSION_GR
 endif()
 set_target_properties(crimson::cflags PROPERTIES
   INTERFACE_COMPILE_DEFINITIONS "${crimson_cflag_definitions}"
+  INTERFACE_COMPILE_OPTIONS "-Wno-non-virtual-dtor"
```
Comment (rzarzynski) [inline src/crimson/CMakeLists.txt]: ACK.

**Comment by tchaikov** [comment]
```
07:06:47 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 42.3301/11.3374  => accepted
07:06:47 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 10836.0/2902.0  => accepted
07:06:47 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.50)):: 767.753/185.236  => rejected
07:06:47 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.00144857/0.00545902  => accepted
07:06:47 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 43.2683/11.4606  => accepted
07:06:47 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 11076.0/2933.0  => accepted
07:06:47 - WARNING  - cbt      - prefill/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.50)):: 625.631/166.64  => rejected
07:06:47 - INFO     - cbt      - prefill/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.00144062/0.0054209  => accepted
07:06:47 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/0: bandwidth: (or (greater) (near 0.05)):: 265.01/52.3609  => accepted
07:06:47 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/0: iops_avg: (or (greater) (near 0.05)):: 67842.0/13404.0  => accepted
07:06:47 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/0: iops_stddev: (or (less) (near 0.50)):: 2369.86/467.211  => rejected
07:06:47 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/0: latency_avg: (or (less) (near 0.05)):: 0.000232733/0.00118389  => accepted
07:06:47 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: bandwidth: (or (greater) (near 0.05)):: 295.82/52.2724  => accepted
07:06:47 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: iops_avg: (or (greater) (near 0.05)):: 75729.0/13381.0  => accepted
07:06:47 - WARNING  - cbt      - rand/incerta08.front.sepia.ceph.com/1: iops_stddev: (or (less) (near 0.50)):: 4234.03/490.271  => rejected
07:06:47 - INFO     - cbt      - rand/incerta08.front.sepia.ceph.com/1: latency_avg: (or (less) (near 0.05)):: 0.000208316/0.00118574  => accepted
07:06:47 - WARNING  - cbt      - 4 tests failed out of 16
```

**Pull request merged**